scripts/python/verify_agent_os.py

```diff
@@ -1,14 +1,26 @@
 # Script verifies the operating system for the platform on which it is being run
 # Used in build pipelines to verify the build agent os
-# Variable: The name of the os to verfy against
-
+# Variable: The friendly name or image name of the os to verfy against
+from __future__ import print_function
 import sys
 import platform
-os_parameter = sys.argv[1]
-print ("Parameter passed by matrix -", os_parameter)
-os_parameter = 'Darwin' if(os_parameter.lower() == 'MacOS'.lower()) else os_parameter
+
+os_parameter = sys.argv[1].lower()
+if os_parameter.startswith('macos'):
+	os_parameter = 'macOS'
+elif os_parameter.startswith('ubuntu'):
+	os_parameter = 'Linux'
+elif os_parameter.startswith('vs') or os_parameter.startswith('windows'):
+	os_parameter = 'Windows'
+
+print("Job requested to run on OS: %s" % (os_parameter))
+
 agent_os = platform.system()
+if agent_os == 'Darwin':
+	agent_os = 'macOS'
+
 if (agent_os.lower() == os_parameter.lower()):
-	print ('Job ran on %s OS' %agent_os)
+	print('Job ran on OS: %s' % (agent_os))
+	print('##vso[task.setvariable variable=OSName]%s' % (os_parameter))
```

Comment:
Could you let me know what this line does? Thanks!

Reply:
Yep! It causes AzDevOps to set a variable (OSName) to a value.